libtransmission/rpcimpl.cc

```diff
@@ -2332,141 +2054,634 @@ void add_strings_from_var(std::set<std::string_view>& strings, tr_variant const&
+            {
+                tr_sessionLimitSpeed(&tgt, TR_DOWN, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_speed_limit_up,
+        [](tr_session const& src) -> tr_variant { return src.speed_limit(TR_UP).count(Speed::Units::KByps); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<int64_t>())
+            {
+                tgt.set_speed_limit(TR_UP, Speed{ *val, Speed::Units::KByps });
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_speed_limit_up_enabled,
+        [](tr_session const& src) -> tr_variant { return src.is_speed_limited(TR_UP); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionLimitSpeed(&tgt, TR_UP, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_start_added_torrents,
+        [](tr_session const& src) -> tr_variant { return !src.shouldPauseAddedTorrents(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetPaused(&tgt, !*val);
+            }
+        });
+
+    map.try_emplace(TR_KEY_tcp_enabled, [](tr_session const& src) -> tr_variant { return src.allowsTCP(); }, nullptr);
+
+    map.try_emplace(
+        TR_KEY_trash_original_torrent_files,
+        [](tr_session const& src) -> tr_variant { return src.shouldDeleteSource(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetDeleteSource(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(TR_KEY_units, [](tr_session const& /*src*/) -> tr_variant { return values_get_units(); }, nullptr);
+
+    map.try_emplace(
+        TR_KEY_utp_enabled,
+        [](tr_session const& src) -> tr_variant { return src.allowsUTP(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetUTPEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_version,
+        [](tr_session const& /*src*/) -> tr_variant { return tr_variant::unmanaged_string(LONG_VERSION_STRING); },
+        nullptr);
+
+    map.try_emplace(
+        TR_KEY_script_torrent_added_enabled,
+        [](tr_session const& src) -> tr_variant { return src.useScript(TR_SCRIPT_ON_TORRENT_ADDED); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tgt.useScript(TR_SCRIPT_ON_TORRENT_ADDED, *val);
+            }
+        });
+
+    for (auto const& row : tr_session::Scripts)
```

Comment:
*(minor nit)*

```suggestion
    for (auto const& [enabled_key, filename_key, script] : tr_session::Scripts)
```

Reply:
That was my first attempt too, but apparently it's not legal in std=c++17 to use a lambda capture of structured bindings.

According to [this page](https://stackoverflow.com/questions/46114214/lambda-implicit-capture-fails-with-variable-declared-from-structured-binding) this is legal in c++20 though